modules/tensor_mechanics/src/materials/cohesive_zone_model/SlipWeakeningFriction3d.C

```diff
@@ -0,0 +1,182 @@
+    _reaction_slipweakening_neighbor_x(coupledNeighborValue("reaction_slipweakening_x")),
+    _reaction_slipweakening_y(coupledValue("reaction_slipweakening_y")),
+    _reaction_slipweakening_neighbor_y(coupledNeighborValue("reaction_slipweakening_y")),
+    _reaction_slipweakening_z(coupledValue("reaction_slipweakening_z")),
+    _reaction_slipweakening_neighbor_z(coupledNeighborValue("reaction_slipweakening_z")),
+    _disp_slipweakening_x_old(coupledValueOld("disp_slipweakening_x")),
+    _disp_slipweakening_neighbor_x_old(coupledNeighborValueOld("disp_slipweakening_x")),
+    _disp_slipweakening_y_old(coupledValueOld("disp_slipweakening_y")),
+    _disp_slipweakening_neighbor_y_old(coupledNeighborValueOld("disp_slipweakening_y")),
+    _disp_slipweakening_z_old(coupledValueOld("disp_slipweakening_z")),
+    _disp_slipweakening_neighbor_z_old(coupledNeighborValueOld("disp_slipweakening_z")),
+    _mu_s(coupledValue("mu_s")),
+    _ini_shear_sts(coupledValue("ini_shear_sts"))
+{
+}
+
+void
+SlipWeakeningFriction3d::computeInterfaceTractionAndDerivatives()
+{
+  // Global Displacement Jump
+  RealVectorValue displacement_jump_global(
+      _disp_slipweakening_x[_qp] - _disp_slipweakening_neighbor_x[_qp],
+      _disp_slipweakening_y[_qp] - _disp_slipweakening_neighbor_y[_qp],
+      _disp_slipweakening_z[_qp] - _disp_slipweakening_neighbor_z[_qp]);
+
+  // Global Displacement Jump Old
+  RealVectorValue displacement_jump_old_global(
+      _disp_slipweakening_x_old[_qp] - _disp_slipweakening_neighbor_x_old[_qp],
+      _disp_slipweakening_y_old[_qp] - _disp_slipweakening_neighbor_y_old[_qp],
+      _disp_slipweakening_z_old[_qp] - _disp_slipweakening_neighbor_z_old[_qp]);
+
+  // Global Displacement Jump Rate
+  RealVectorValue displacement_jump_rate_global =
+      (displacement_jump_global - displacement_jump_old_global) * (1 / _dt);
+
+  // Local Displacement Jump / Displacement Jump Rate
+  RealVectorValue displacement_jump = _rot[_qp].transpose() * displacement_jump_global;
+  RealVectorValue displacement_jump_rate = _rot[_qp].transpose() * displacement_jump_rate_global;
+
+  // Parameter initialization
+  Real mu_s = 0;
+  Real mu_d = _mu_d;
+  Real Dc = _Dc;
+  Real tau_f = 0;
+
+  Real T1_o = 0;
+  Real T2_o = _T2_o;
+  Real T3_o = _T3_o;
+
+  Real len = _len;
+
+  // Reaction force in local coordinate
+  RealVectorValue R_plus_global(-_reaction_slipweakening_x[_qp],
+                                -_reaction_slipweakening_y[_qp],
+                                -_reaction_slipweakening_z[_qp]);
+  RealVectorValue R_minus_global(-_reaction_slipweakening_neighbor_x[_qp],
+                                 -_reaction_slipweakening_neighbor_y[_qp],
+                                 -_reaction_slipweakening_neighbor_z[_qp]);
+
+  RealVectorValue R_plus_local = _rot[_qp].transpose() * R_plus_global;
+  RealVectorValue R_minus_local = _rot[_qp].transpose() * R_minus_global;
+
+  Real R_plus_local_x = R_plus_local(1);
```

Comment:
Is there a reason for giving x an index of 1 instead of 0?

Reply:
use a variable instead of 1 , 0 , 2